tools/debug/debugsymbolviewer.py

```diff
@@ -82,21 +114,57 @@ def setup_symbol_table(sm_file):
 		print('~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~')
 		print('')
 
+# Function to setup Address to Symbol mapping table from the System Map and appX map files for kernel and applications
+def setup_symbol_table(map_file, is_app):
+	if debug:
+		print('setup_symbol_table : '+ map_file, is_app)
+	# Reading the System Map file and preparing the symbol map table
+	with open(map_file) as searchfile:
+		for line in searchfile:
+			s = line.split(' ', 2)
+			if len(s) == 3:
+				if not '$' in s[2]:
+					# s[0] contains address and s[2] contains Symbol name
+					if(is_app):
+						asymbol_lookup_table[is_app].append((int(s[0], 16), s[2].rstrip()))
+					else:
+						ksymbol_lookup_table.append((int(s[0], 16), s[2].rstrip()))
+	if debug:
+		# Printing the Created Symbol table
+		print('\n~~~~~~~~~~~~~~~~~~~~~~~~ SYMBOL TABLE START ~~~~~~~~~~~~~~~~~~~~~')
+		if(is_app):
+			for line in asymbol_lookup_table[is_app]:
+				print("{0:x} {1}".format(line[0], line[1]))
+		else:
+			for line in ksymbol_lookup_table:
+				print("{0:x} {1}".format(line[0], line[1]))
+		print('~~~~~~~~~~~~~~~~~~~~~~~~SYMBOL TABLE END   ~~~~~~~~~~~~~~~~~~~~~')
 
-# Function to search the search_addr present in the System map file and print its corresponding symbol
-def print_symbol(search_addr):
+
+# Function to search the search_addr present in the Address to Symbol mapping table and print its corresponding symbol
+def print_symbol(stack_addr, search_addr, is_app_symbol):
 	addr = 0x00000000
 	next_addr = 0x00000000
 	l = 0
-	h = len(symbol_lookup_table) - 1
+	if (is_app_symbol):
+		h = len(asymbol_lookup_table[is_app_symbol]) - 1
```

Comment:
`is_app_symbol` has an index + 1, which returned from `is_app_text_address`.
I think we should use "`is_app_symbol` - 1" for using as the index of `asymbol_lookup_table`.

Reply:
Since the symbol table generation is based upon the application indexes for apps & 0 for kernel symbol table as below:
```
    setup_symbol_table(bin_path + "System.map", 0)
    for idx in range(app_idx):
        os.system("nm --defined-only -l --numeric-sort " + bin_path + app_name[idx] + "_dbg > " + bin_path + app_name[idx] + ".map")
        setup_symbol_table(bin_path + app_name[idx] + ".map", idx + 1)
```
The index of 'asymbol_lookup_table' is correct for the given implementation.